### PR TITLE
feat: adding reverse transactions

### PR DESCRIPTION
**Pull request by vaibhawvipul**
- adding reverse scan
- for `Transaction::Scan` and `DB::Scan`
- added test cases

### CODE REVIEW

**Review by ethe — CHANGES_REQUESTED** [review]
Because Tonbo based on LSM Tree, mutable and immutable memtable is just the in-memory part of storage, I think `LevelStream` also needs to be supported. number of records in test cases is too small, which means all records are stored in mutable membtale, behaviors of sstable part are missed.

**Comment by ethe** [comment]
Also `test_reverse_scan_with_struct` is not correct, because just reverse lower bound and upper bound does not make sense to cross_beam `SkipMap`, it does not support this to reverse the iteration, you need to change the behavior of `MutableScan`.

File: src/lib.rs [hunk 1]
```diff
@@ -587,39 +587,55 @@ where
...
-                .mutable
-                .scan((self.lower, self.upper), self.ts)
-                .into(),
-        );
-        for (_, immutable) in self.schema.immutables.iter() {
-            let (lower, upper) = if self.reversed == Some(true) {
+        // Helper function to handle reversing logic
+        fn maybe_reverse<T>(vec: &mut Vec<T>, reverse: bool) {
+            if reverse {
+                vec.reverse();
```
Comment (ethe) [inline src/lib.rs:593]: reverse the vector of merge stream dooes not make sense, the order of merge stream should keep `mutable memtable -> immutable memtable -> leveled sstable` to ensure read the value with highest timestamp.

Comment (vaibhawvipul) [reply src/lib.rs:593]: understood.

**Comment by ethe** [comment]
close for stale， thanks vipul!

**Pull request closed without merging**